File: apps/api/internal/model/travel.go
package model

type Trip struct {
	ID        string   `json:"id"`
	OwnerID   string   `json:"ownerId"`
	Title     string   `json:"title"`
	StartDate string   `json:"startDate"`
	EndDate   string   `json:"endDate"`
	Travelers []string `json:"travelers"`
	Memo      string   `json:"memo,omitempty"`
}

type Schedule struct {
	ID            string `json:"id"`
	TripID        string `json:"tripId"`
	PlaceID       string `json:"placeId,omitempty"`
	Date          string `json:"date"`
	Time          string `json:"time"`
	Type          string `json:"type"`
	Title         string `json:"title"`
	TransportMemo string `json:"transportMemo,omitempty"`
	GuideMemo     string `json:"guideMemo,omitempty"`
}

type Place struct {
	ID                string `json:"id"`
	TripID            string `json:"tripId"`
	Name              string `json:"name"`
	Category          string `json:"category"`
	Address           string `json:"address,omitempty"`
	GoogleMapsURL     string `json:"googleMapsUrl,omitempty"`
	RecommendedReason string `json:"recommendedReason,omitempty"`
}

type Flight struct {
	ID               string `json:"id"`
	TripID           string `json:"tripId"`
	Direction        string `json:"direction"`
	Label            string `json:"label"`
	Airline          string `json:"airline,omitempty"`
	FlightNumber     string `json:"flightNumber,omitempty"`
	DepartureAirport string `json:"departureAirport"`
	ArrivalAirport   string `json:"arrivalAirport"`
	DepartureDate    string `json:"departureDate"`
	DepartureTime    string `json:"departureTime"`
	ArrivalDate      string `json:"arrivalDate,omitempty"`
	ArrivalTime      string `json:"arrivalTime,omitempty"`
	Memo             string `json:"memo,omitempty"`
}

type Route struct {
	ID                string   `json:"id"`
	TripID            string   `json:"tripId"`
	Title             string   `json:"title"`
	Description       string   `json:"description,omitempty"`
	PlaceIDs          []string `json:"placeIds"`
	TransportMemo     string   `json:"transportMemo,omitempty"`
	EstimatedDuration string   `json:"estimatedDuration,omitempty"`
}

File: apps/web/src/types/travel.ts
export type ScheduleType = "move" | "meal" | "golf" | "sightseeing" | "hotel" | "shopping" | "etc";

export type PlaceCategory =
  | "hotel"
  | "meal"
  | "golf"
  | "cafe"
  | "sightseeing"
  | "shopping"
  | "transport"
  | "etc";

export type Trip = {
  title: string;
  startDate: string;
  endDate: string;
  travelers: string[];
  destinationCountry?: string;
  memo?: string;
};

export type FlightInfo = {
  id: string;
  label: string;
  airline?: string;
  flightNumber?: string;
  date?: string;
  time?: string;
  memo?: string;
};

export type AccommodationInfo = {
  name: string;
  address: string;
  phone?: string;
  checkIn?: string;
  checkOut?: string;
  memo?: string;
};

export type UsefulPhrase = {
  id: string;
  situation: string;
  korean: string;
  japanese?: string;
  chinese?: string;
};

export type ScheduleItem = {
  id: string;
  date: string;
  time: string;
  type: ScheduleType;
  title: string;
  placeId?: string;
  transportMemo?: string;
  reservationMemo?: string;
  guideMemo?: string;
};

export type Place = {
  id: string;
  name: string;
  category: PlaceCategory;
  address?: string;
  latitude?: number;
  longitude?: number;
  googleMapsUrl?: string;
  recommendedReason?: string;
  openingMemo?: string;
  budgetMemo?: string;
  cautionMemo?: string;
};

export type RecommendedRoute = {
  id: string;
  title: string;
  description?: string;
  placeIds: string[];
  transportMemo?: string;
  estimatedDuration?: string;
};

export type ChecklistItem = {
  id: string;
  category: "before" | "airport" | "daily" | "return";
  title: string;
  destinationCountry?: string;
  memo?: string;
};

export type EmergencyInfo = {
  id: string;
  title: string;
  description: string;
  phone?: string;
  address?: string;
};
